Parameters 'client_id' and 'response_type' not strictly required in O…

mposolda opened the pull request:
…IDC request object

Closes #14255

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

mposolda committed:
Parameters 'client_id' and 'response_type' not strictly required in OIDC request object
Closes #14255

tnorimat commented on testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/OIDCAdvancedRequestParamsTest.java:
```diff
@@ -631,6 +631,56 @@ public void requestObjectSupersedesQueryParameter() throws Exception {
+        errorPage.assertCurrent();
+
+        // Test that "response_type" mandatory in the query even if set in the "request" object
+        oidcClientEndpointsResource.setOIDCRequest("test", "test-app", oauth.getRedirectUri(), "10", "some-state", "none");
```
This line and the following line might not be needed.

mposolda replied:
Thanks for catching this. I've removed those lines. I've used separate commit, just so you can easily re-review that just this was changed. I plan to squash the commits before merging this.

tnorimat reviewed (commented):
@mposolda I added one review comment. Could you check it?

tnorimat commented:
@mposolda I ran FAPI1 advanced final conformance test against this PR and confirmed that the PR can pass all conformance tests. I will run OIDC conformance tests and others against the PR.

tnorimat commented:
@mposolda I ran other conformance tests (OIDC, OIDC for Logout Profiles, FAPI-CIBA, Open Banking Brazil FAPI ID2, Australia CDR, UK Open Banking) and confirmed that the PR can pass all conformance tests.

mposolda committed:
Feedback in the PR

mposolda commented:
@tnorimat Thanks for the review and for doublecheck the FAPI and OIDC testsuite! I've addressed your comment in the OIDCAdvancedRequestParamsTest. Can you please re-review if looks ok now?

tnorimat commented:
@mposolda Yes, I will re-review the PR.

tnorimat reviewed (approved):
LGTM
